vendor/engines/secure_rooms/db/migrate/20170316090613_add_unique_index_to_card_readers.rb

```diff
@@ -0,0 +1,12 @@
+class AddUniqueIndexToCardReaders < ActiveRecord::Migration
+
+  def change
+    add_index(
+      :secure_rooms_card_readers,
+      [:product_id, :card_reader_number, :control_device_number],
```

Comment:
I might need to think about this a little more, but should `[:card_reader_number, :control_device_number`]` be globally unique, not just within the product?